### PR TITLE
MBS-7854 update kotlin 1.3.61 -> 1.3.71

### PR DESCRIPTION
**Pull request by dsvoronin**
https://blog.jetbrains.com/kotlin/2020/03/kotlin-1-3-70-released/

Previous PR's clashed with new Dependencies.kt, and effective version was still 1.3.61

Still an AssertionError problem from last PR #220 

```
> Task :subprojects:android-test:test-app:inferStagingAndroidTestUsageForKeeper FAILED
Exception in thread "main" java.lang.AssertionError
        at com.android.tools.r8.PrintUses$UseCollector.addMethod(PrintUses.java:211)
        at com.android.tools.r8.PrintUses$UseCollector.registerInvokeStatic(PrintUses.java:108)
        at com.android.tools.r8.cf.code.CfInvoke.registerUse(CfInvoke.java:101)
        at com.android.tools.r8.graph.CfCode.registerCodeReferences(CfCode.java:376)
        at com.android.tools.r8.graph.LazyCfCode.registerCodeReferences(LazyCfCode.java:204)
        at com.android.tools.r8.graph.DexEncodedMethod.registerCodeReferences(DexEncodedMethod.java:1070)
        at com.android.tools.r8.PrintUses$UseCollector.registerMethod(PrintUses.java:240)
        at com.android.tools.r8.PrintUses$UseCollector.access$1200(PrintUses.java:77)
        at com.android.tools.r8.PrintUses.lambda$analyze$0(PrintUses.java:340)
        at com.android.tools.r8.graph.DexClass.forEachMethod(DexClass.java:294)
        at com.android.tools.r8.PrintUses.analyze(PrintUses.java:340)
        at com.android.tools.r8.PrintUses.main(PrintUses.java:309)
```

### CODE REVIEW

**Comment by dsvoronin** [comment]
ok, finally found some clues: 
bumping r8 version to 2.0.46 adds more error info:
`Could not find method Method kotlin.reflect.KClasses.getDefaultType Proto LL kotlin.reflect.KType kotlin.reflect.KClass`

**Comment by dsvoronin** [comment]
issues created: 
slackhq/keeper/issues/31
https://issuetracker.google.com/issues/151002499

File: settings.gradle.kts [hunk 1]
```diff
@@ -156,7 +154,7 @@ pluginManagement {
                     useModule("com.android.tools.build:gradle:3.5.3")
 
                 pluginId.startsWith("org.jetbrains.kotlin.") ->
-                    useVersion(kotlinVersion)
+                    useVersion("1.3.70")
```
Comment (sergioboi) [inline settings.gradle.kts]: Why do we need that hardcode?

Comment (dsvoronin) [reply settings.gradle.kts]: Dependencies.kt is not available in settings.gradle.kts

Comment (sergioboi) [inline settings.gradle.kts]: Do we have access to Dependencies.kt?

File: buildSrc/src/main/java/Dependencies.kt [hunk 2]
```diff
@@ -4,7 +4,7 @@ object Dependencies {
         val okhttp = "3.14.6"
         val sentry = "1.7.23"
         val retrofit = "2.6.2"
-        val kotlin = "1.3.61"
+        val kotlin = "1.3.70"
```
Comment (sergioboi) [inline buildSrc/src/main/java/Dependencies.kt]: Is our goal to delete this?

File: settings.gradle.kts (hunk 1 above)
Comment (dsvoronin) [reply settings.gradle.kts]: no afaik

File: buildSrc/src/main/java/Dependencies.kt (hunk 2 above)
Comment (dsvoronin) [reply buildSrc/src/main/java/Dependencies.kt]: we should try, kotlin plugin provides "platform" data

**Comment by dsvoronin** [comment]
https://github.com/JetBrains/kotlin/releases/tag/v1.3.71

File: settings.gradle.kts (hunk 1 above)
Comment (dsvoronin) [reply settings.gradle.kts]: https://docs.gradle.org/current/userguide/upgrading_version_5.html#classes_from_buildsrc_are_no_longer_visible_to_settings_scripts

**Comment by dsvoronin** [comment]
it was mockito-kotlin 1.5 with bundled kotlin-reflect classes

**Comment by sergioboi** [comment]
> it was mockito-kotlin 1.5 with bundled kotlin-reflect classes

BIATCH

File: buildSrc/build.gradle.kts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 plugins {
-    id("org.jetbrains.kotlin.jvm") version "1.3.61"
+    `kotlin-dsl`
```
Comment (sergioboi) [inline buildSrc/build.gradle.kts:2]: Does buildSrc see systemProp.kotlinVersion or how does it work?

**Review by sergioboi — APPROVED** [review]

**Pull request merged**